src/scimPatch.ts

```diff
@@ -122,7 +122,7 @@ function applyAddOperation(scimResource: ScimResource, patch: ScimPatchAddReplac
         return addOrReplaceAttribute(scimResource, patch);
 
     // We navigate till the second to last of the path.
-    const paths = patch.path.split('.');
+    const paths = patch.path.split(/(?!\d)\.(?!\d)/g);
```

Comment:
To avoid duplication on the regex, could you but this is a constant?
You could put it under the line
```
const ARRAY_SEARCH: RegExp = /^(.+)\[(.+)\]$/;
```
at the top of the file.

Reply:
Done! :)